File: src/node.cc
#include "node.h"

template <typename T>

Node::Node(T val) : data(val), next(nullptr) {}

Node::Node(T val) : data(val), next(nullptr) {}

Node::Node() : data(0), next(nullptr) {}

T Node::getData() { return data; }

void Node::setData(T val) { data = val; }

Node *Node::getNext() { return next; }

void Node::setNext(Node *node) { next = node; }

~Node::Node() {}

File: Data-Structures/Linked-Lists/node.py
class Node:
    def __init__(self, value, next = None):
        self.value = value
        self.next = next

    def getValue(self):
        return self.value
    
    def setValue(self, value):
        self.value = value
    
    def getNext(self):
        return self.next

    def setNext(self, next):
        self.next = next
